src/player.js

```diff
@@ -1223,15 +1240,38 @@ export default class Player extends FakeEventTarget {
    */
   _setDefaultTracks(): void {
     const activeTracks = this.getActiveTracks();
-    const playbackConfig = this._config.playback;
+    const playbackConfig = this.config.playback;
+    const offTextTrack: ?Track =  this._getTracksByType(TrackTypes.TEXT).find(track => TextTrack.langComparer(OFF, track.language));
 
     this.hideTextTrack();
 
-    const textLanguage = (playbackConfig.textLanguage === "auto") ? Locale.language : playbackConfig.textLanguage;
-    this._setDefaultTrack(TrackTypes.TEXT, textLanguage, activeTracks.text);
+    this._setDefaultTrack(TrackTypes.TEXT, this._getTextLanguage(activeTracks, playbackConfig.textLanguage), offTextTrack);
     this._setDefaultTrack(TrackTypes.AUDIO, playbackConfig.audioLanguage, activeTracks.audio);
   }
 
+
+  /**
+   * Gets the text value that should be set in the captions - from the browser / activeTrack if the confiuuration is Auto
+   * @param {Object} activeTracks - active track data
+   * @param {string} textLanguage - the language, in case the configuration is not auto
+   * @private
+   * @returns {string}
+   */
+
+  _getTextLanguage(activeTracks: Object, textLanguage: string): string {
```

Comment:
better call this `_getLanguage` and pass it the `activeTracks.text`. 
This way we can use it generically for both text and audio.

Also remember to change the flow type check to be of type Track instead of Object.

Reply:
Need to check what happens for default audio as well.
The `Auto` should apply for it as well.
Does audio always come back with `activeTracks.audio` after parsing the manifest? if so the Locale option won't be valid in any case - @odedhutzler need to check this with both HLS on Safari, HLS with HLS.JS and Dash with Shaka.